internal/loader/configuration_setting_loader.go

```diff
@@ -743,3 +758,40 @@ func MergeSecret(secret map[string]corev1.Secret, newSecret map[string]corev1.Se
 
 	return nil
 }
+
+func createCorrelationContextHeader(provider acpv1.AzureAppConfigurationProvider) http.Header {
+	header := http.Header{}
+	output := make([]string, 0)
+
+	if provider.Spec.Configuration.Refresh != nil {
```

Comment:
`Spec.Configuration` can be nil

Reply:
`Refresh.Enabled`  can be false, need to check it when `Configuration.Refresh` section has been set.